packages/medusa-payment-stripe/src/api/hooks/stripe.ts

```diff
@@ -0,0 +1,161 @@
+import { Request, Response } from "express"
+import {
+  buildHandleCartPaymentErrorMessage,
+  constructWebhook,
+  isPaymentCollection,
+} from "../utils/utils"
+import {
+  AbstractCartCompletionStrategy,
+  CartService,
+  IdempotencyKeyService,
+} from "@medusajs/medusa"
+import { MedusaError } from "medusa-core-utils"
+
+export default async (req: Request, res: Response) => {
+  let event
+  try {
+    event = constructWebhook({
+      signature: req.headers["stripe-signature"],
+      body: req.body,
+      container: req.scope,
+    })
+  } catch (err) {
+    res.status(400).send(`Webhook Error: ${err.message}`)
+    return
+  }
+
+  const paymentIntent = event.data.object
+  const cartId = paymentIntent.metadata.cart_id // Backward compatibility
+  const resourceId = paymentIntent.metadata.resource_id
+
+  if (isPaymentCollection(resourceId)) {
+    await handlePaymentCollection(event, req, res, resourceId, paymentIntent.id)
+  } else {
+    await handleCartPayments(event, req, res, cartId ?? resourceId)
+  }
+}
+
+async function handleCartPayments(event, req, res, cartId) {
+  const manager = req.scope.resolve("manager")
+  const orderService = req.scope.resolve("orderService")
+  const logger = req.scope.resolve("logger")
+
+  const order = await orderService
+    .retrieveByCartId(cartId)
+    .catch(() => undefined)
+
+  // handle payment intent events
+  switch (event.type) {
+    case "payment_intent.succeeded":
+      if (order) {
+        // If order is created but not captured, we attempt to do so
+        if (order.payment_status !== "captured") {
+          await manager.transaction(async (manager) => {
+            await orderService.withTransaction(manager).capturePayment(order.id)
+          })
+        } else {
+          // Otherwise, respond with 200 preventing Stripe from retrying
+          return res.sendStatus(200)
+        }
+      } else {
+        // If order is not created, we respond with 404 to trigger Stripe retry mechanism
+        return res.sendStatus(404)
+      }
```

Comment:
suggestion(non-blocking): Extract into a method to be consistent with the other case